src/v/http/client.cc

```diff
@@ -639,6 +640,23 @@ ss::future<client::response_stream_ref> client::request(
     return request(std::move(header), iobuf(), timeout);
 }
 
+ss::future<http::downloaded_response> client::request_and_collect_response(
+  request_header&& req,
+  std::optional<iobuf> payload,
+  ss::lowres_clock::duration timeout) {
+    response_stream_ref response;
+    if (payload.has_value()) {
+        response = co_await request(
+          std::move(req), std::move(payload.value()), timeout);
+    } else {
+        response = co_await request(std::move(req), timeout);
+    }
+
+    co_return downloaded_response{
+      .status = co_await http::status(response),
+      .body = co_await http::drain(response)};
```

Comment:
I know the order here is that the status is grabbed first, but can we separate this out for my brain to handle? Something like:

```
auto status = co_await http::status(response);
auto body = co_await http::status(response);
co_return downloaded_response{.status = status, .body = std::move(body)};
```

Reply:
done